### PR TITLE
feat: ng-add schematics

### LINKED ISSUE #425: Create a ngAdd Schematics
It could be easier to integrate this package with a schematic

### PR DESCRIPTION
**Pull request by vscaiceanu-1a**
Closes #425

### CODE REVIEW

File: src/schematics/ng-add/index.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { NgAddSchematicsSchema } from './schema';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { chain, Rule, SchematicContext, SchematicsException, Tree } from '@angular-devkit/schematics';
+import { addPackageJsonDependency, getPackageJsonDependency, NodeDependencyType } from '@schematics/angular/utility/dependencies';
+import * as semver from 'semver';
+
+export function install(tree: Tree, context: SchematicContext): Tree {
```
Comment (kpanot) [inline src/schematics/ng-add/index.ts:7]: missing comments

File: src/schematics/ng-add/index.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+function updatePrefetchBuilder(options: NgAddSchematicsSchema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+
+    const workspace = readAngularJson(tree);
+    const projectName = getFinalProjectName(workspace, options.projectName);
+
+    const workspaceProject = workspace.projects[projectName];
+    const packageJson = readPackageJson(tree, workspaceProject);
+
+    if (!packageJson.scripts) {
```
Comment (kpanot) [inline src/schematics/ng-add/index.ts:54]: `packageJson.scripts ||= {}`

Comment (vscaiceanu-1a) [reply src/schematics/ng-add/index.ts]: done, as well for `workspaceProject.architect ||= {};` below

File: src/schematics/ng-add/index.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    tree.overwrite('/angular.json', JSON.stringify(workspace, null, 2));
+
+    return tree;
+  };
+}
+
+
+
+/**
+ * Add Otter library to an Angular Project
```
Comment (kpanot) [inline src/schematics/ng-add/index.ts:71]: Wrong comment?

Comment (vscaiceanu-1a) [reply src/schematics/ng-add/index.ts:71]: nice catch

**Review by kpanot — APPROVED** [review]

**Pull request merged**